ionoscloud/resource_server.go

```diff
@@ -568,65 +568,62 @@ func resourceServerCreate(ctx context.Context, d *schema.ResourceData, meta inte
 
 	serverType := d.Get("type").(string)
 
+	serverReq.Entities = ionoscloud.NewServerEntities()
+
 	// create volume object with data to be used for image
 	volume, err := getVolumeData(d, "volume.0.", serverType)
-
-	if err != nil {
-		return diag.FromErr(err)
-	}
-
-	// get image and imageAlias
-	image, imageAlias, err := getImage(ctx, client, d, *volume)
 	if err != nil {
 		return diag.FromErr(err)
 	}
+	if volume.Type != nil && *volume.Type != "" {
+		// get image and imageAlias
+		image, imageAlias, err := getImage(ctx, client, d, *volume)
+		if err != nil {
+			return diag.FromErr(err)
+		}
 
-	// add remaining properties in volume (dependent in image and imageAlias)
-	if imageAlias != "" {
-		volume.ImageAlias = &imageAlias
-	} else {
+		// add remaining properties in volume (dependent in image and imageAlias)
 		volume.ImageAlias = nil
-	}
+		if imageAlias != "" {
+			volume.ImageAlias = &imageAlias
+		}
 
-	if image != "" {
-		volume.Image = &image
-	} else {
 		volume.Image = nil
-	}
+		if image != "" {
+			volume.Image = &image
+		}
 
-	if backupUnitId, ok := d.GetOk("volume.0.backup_unit_id"); ok {
-		if utils.IsValidUUID(backupUnitId.(string)) {
-			if image == "" && imageAlias == "" {
-				diags := diag.FromErr(fmt.Errorf("it is mandatory to provide either public image or imageAlias in conjunction with backup unit id property"))
-				return diags
-			} else {
-				backupUnitId := backupUnitId.(string)
-				volume.BackupunitId = &backupUnitId
+		if backupUnitID, ok := d.GetOk("volume.0.backup_unit_id"); ok {
+			if utils.IsValidUUID(backupUnitID.(string)) {
+				if image == "" && imageAlias == "" {
+					diags := diag.FromErr(fmt.Errorf("it is mandatory to provide either public image or imageAlias in conjunction with backup unit id property"))
+					return diags
+				}
+				backupUnitID := backupUnitID.(string)
+				volume.BackupunitId = &backupUnitID
 			}
 		}
-	}
 
-	if userData, ok := d.GetOk("volume.0.user_data"); ok {
-		if image == "" && imageAlias == "" {
-			diags := diag.FromErr(fmt.Errorf("it is mandatory to provide either public image or imageAlias that has cloud-init compatibility in conjunction with backup unit id property "))
-			return diags
-		} else {
+		if userData, ok := d.GetOk("volume.0.user_data"); ok {
+			if image == "" && imageAlias == "" {
+				diags := diag.FromErr(fmt.Errorf("it is mandatory to provide either public image or imageAlias that has cloud-init compatibility in conjunction with backup unit id property "))
+				return diags
+			}
 			userData := userData.(string)
 			volume.UserData = &userData
-		}
-	}
 
-	// add volume object to serverReq
-	serverReq.Entities = &ionoscloud.ServerEntities{
-		Volumes: &ionoscloud.AttachedVolumes{
-			Items: &[]ionoscloud.Volume{
-				{
-					Properties: volume,
+		}
+		// add volume object to serverReq
+		serverReq.Entities = &ionoscloud.ServerEntities{
```

Comment:
@cristiGuranIonos since `serverReq.Entities` is instantiated above: `571: serverReq.Entities = ionoscloud.NewServerEntities()` I think we can just populate the `Volumes` property here.

Reply:
done